server/channels/api4/channel.go

```diff
@@ -60,6 +60,8 @@ func (api *API) InitChannel() {
 	api.BaseRoutes.Channel.Handle("/members_minus_group_members", api.APISessionRequired(channelMembersMinusGroupMembers)).Methods("GET")
 	api.BaseRoutes.Channel.Handle("/move", api.APISessionRequired(moveChannel)).Methods("POST")
 	api.BaseRoutes.Channel.Handle("/member_counts_by_group", api.APISessionRequired(channelMemberCountsByGroup)).Methods("GET")
+	api.BaseRoutes.Channel.Handle("/common_teams", api.APISessionRequired(getGroupMessageMembersCommonTeams)).Methods("GET")
+	api.BaseRoutes.Channel.Handle("/convert_to_channel", api.APISessionRequired(convertGroupMessageToChannel)).Methods("POST")
```

Comment:
We could. It would remove the REST endpoint but everything being done in the app layer for this operation will become a large `if` block in patch channel. Like if type was G and new is P, do all this stuff, and maybe nothing else.

This also feels more explicit to me.

Any issue do you see with separate endpoint, or any benefit of doing it in patch channel endpoint?

Reply:
LGTM. Just keeping the API tidy. I think we used to have a API to convert a channel to private that was removed at some point and broke the mobile app, so I didn't want for that to happen again :P

But your reasoning looks good to me.

Reply:
👍